build.gradle

```diff
@@ -1,22 +1,182 @@
-apply plugin: 'java'
-apply plugin: 'checkstyle'
-
-dependencies {
-    testImplementation("org.junit.jupiter:junit-jupiter:5.8.2")
-}
-
-test {
-    useJUnitPlatform()
-}
-
-task Checkstyle(type: Checkstyle) {
-    configFile = rootProject.file('config/checkstyle/checkstyle.xml')
-    source 'src/main/java'
-    include '**/*.java'
-
-    // empty classpath
-    classpath = files()
-
-}
-
-repositories { mavenCentral() }
\ No newline at end of file
+//Build script is slightly modified to automatically incorporate the parchment mappings.
+buildscript {
+    repositories {
+        // These repositories are only for Gradle plugins, put any other repositories in the repository block further below
+        maven { url = 'https://maven.minecraftforge.net' }
+        maven { url = 'https://maven.parchmentmc.org' }
+        mavenCentral()
+    }
+    dependencies {
+        classpath group: 'net.minecraftforge.gradle', name: 'ForgeGradle', version: '5.1.+', changing: true
+        classpath 'org.parchmentmc:librarian:1.+'
+    }
+}
+apply plugin: 'net.minecraftforge.gradle'
+apply plugin: 'org.parchmentmc.librarian.forgegradle'
+// Only edit below this line, the above code adds and enables the necessary things for Forge to be setup.
+apply plugin: 'eclipse'
+apply plugin: 'maven-publish'
+apply plugin: 'java'
+apply plugin: 'checkstyle'
+
+version = '0.1'
+group = 'team.ocelot.birdhaus' // http://maven.apache.org/guides/mini/guide-naming-conventions.html

```

Comment:
I think version also needs to follow this convention according to forge documentation:
MinecraftVersion-ModMajorVersion.ModMinorVersion.ModPatchVersion

So it would be 1.18-0.1.0

Reply:
@chris7neves could you drop a link to the documentation

Reply:
https://mcforge.readthedocs.io/en/latest/conventions/versioning/

Seems like kind of a "nice to have" way to do it.

Here's the paragraph im referring to: "
However, in the case of modding it may be more beneficial to use the format MCVERSION-MAJORMOD.MAJORAPI.MINOR.PATCH to be able to differentiate between world-breaking and API-breaking changes of a mod.
"

Reply:
im convinced :) i think we should do this